src/viewer.ts

```diff
@@ -524,6 +535,32 @@ export class SkinViewer {
 		}
 	}
 
+	private lastFrameTime: number = 0;
+	private capeSource?: TextureSource;
+	private capeOptions: CapeLoadOptions = {}
+	private totalFrames = 1;
+	private currentFrame = 1;
+	animatedCape(): void {
+		const currentTime: number = Date.now();
+		if(this.lastFrameTime + +this._frameDelay > currentTime) return;
+		this.lastFrameTime = currentTime
+
+		if(this.capeSource != null && this.totalFrames > 1) {
+			this.loadCape(this.capeSource, this.capeOptions)
```

Comment:
Calling `loadCape()` on each frame update would have a huge performance impact. I think it's better to load each frame into a individual canvas and just switch the canvas periodically.